src/main/scala/is/hail/expr/ir/Compile.scala

```diff
@@ -98,3 +110,119 @@ object Compile {
     ), body)
   }
 }
+
+object CompileWithAggregators {
+  def apply[
+    F1 >: Null : TypeInfo,
+    F2 >: Null : TypeInfo,
+    R: TypeInfo : ClassTag
+  ](aggName: String,
+    aggType: TAggregable,
+    args: Seq[(String, Type, ClassTag[_])],
+    aggScopeArgs: Seq[(String, Type, ClassTag[_])],
+    body: IR
+  ): (Array[RegionValueAggregator], Array[() => F1], Type, () => F2, Type) = {
+
+    assert((args ++ aggScopeArgs).forall { case (_, t, ct) => TypeToIRIntermediateClassTag(t) == ct })
+
+    val env = args.zipWithIndex
+      .foldLeft(Env.empty[IR]) { case (e, ((n, t, _), i)) => e.bind(n, In(i, t)) }
```

Comment:
I think this should still shift the index over by one, so that the first argument is In(1), and In(0) is the aggregable